craft_providers/base.py

```diff
@@ -655,6 +655,7 @@ def _install_snaps(self, executor: Executor) -> None:
                     ),
                 )
 
+            # this is only used when environment is set.
```

Comment:
I'm not sure if this belongs inside craft-providers. It explains business logic about how a craft app could decide to set this parameter.

```suggestion
```